openmls/src/credentials/mod.rs

```diff
@@ -212,31 +333,50 @@ pub struct CredentialBundle {
 }
 
 impl CredentialBundle {
-    /// Creates and returns a new [`CredentialBundle`] of the given
-    /// [`CredentialType`] for the given identity and [`SignatureScheme`]. The
-    /// corresponding key material is freshly generated.
-    ///
-    /// Returns an error if the given [`CredentialType`] is not supported.
-    pub fn new(
+    /// Creates and returns a new basic [`CredentialBundle`] for the given identity and [`SignatureScheme`].
+    /// The corresponding key material is freshly generated.
+    pub fn new_basic(
```

Comment:
I don't think it's worth breaking the api for that
Maybe just going to 2 private functions (`new_basic` and `new_x509`) and keeping the current api that matches over `credential_type` and goes to the appropriate internal constructor

Reply:
Here, both methods will diverge and ultimately have no parameters in common (`identity` in `new_certificate` will go away soon).
`new_certificate` also requires 2 new parameters. So anyway we'll have an API break. But keeping a single `new` method would mean having its 5 parameters wrapped in `Option` and user carefully picking which one to set depending if he wants to build a basic or a certificate `CredentialBundle`